ape_safe/client.py

```diff
@@ -220,17 +275,88 @@ def post_transaction(self, safe_tx: SafeTx, sigs: Optional[List[MessageSignature
             )
 
         url = f"{self.transaction_service_url}/api/v1/multisig-transactions"
-        response = requests.post(url, json=tx_data.dict())
+        response = requests.post(url, json={"origin": "ApeWorX/ape-safe", **tx_data.dict()})
 
         if not response.ok:
-            raise
+            raise ClientResponseError(url, response)
 
-    def post_signature(self, safe_tx_id: SafeTxID, signature: MessageSignature):
+    def post_signature(self, safe_tx_hash: SafeTxID, signature: MessageSignature):
         url = (
-            f"{self.transaction_service_url}"
-            f"/api/v1/multisig-transactions/{safe_tx_id.hex()}/confirmations"
+            f"{self.transaction_service_url}/api"
+            f"/v1/multisig-transactions/{str(safe_tx_hash)}/confirmations"
+        )
+        response = requests.post(
+            url, json={"origin": "ApeWorX/ape-safe", "signature": signature.encode_vrs().hex()}
```

Comment:
the correct encoding is rsv
```suggestion
            url, json={"origin": "ApeWorX/ape-safe", "signature": signature.encode_rsv().hex()}
```